server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferWithLinkedList.java

```diff
@@ -110,8 +110,26 @@ public int getBlockCount() {
   }
 
   @Override
-  public void release() {
-    blocks.forEach(spb -> spb.getData().release());
+  public long release() {
+    Throwable lastException = null;
+    int failedReleaseSize = 0;
+    long releaseSize = 0;
+    for (ShufflePartitionedBlock spb : blocks) {
+      try {
+        spb.getData().release();
+        releaseSize += spb.getSize();
+      } catch (Throwable t) {
+        lastException = t;
```

Comment:
Should we log here?

Reply:
Yeah, but it can be verbose, may be we get last exception is enough?

Reply:
Think twice. It's ok.